jdaviz/app.py

```diff
@@ -1324,34 +1348,15 @@ def merge_overlapping_spectral_regions(self, subset_name, att):
         att : str
             Attribute that the subset uses to apply to data.
         """
-        spectral_region = self.get_subsets(subset_name, spectral_only=True)
-        merged_regions = None
-        # Convert SpectralRegion object into a list with tuples representing
-        # the lower and upper values of each region.
-        reg_as_tup = [(sr.lower.value, sr.upper.value) for sr in spectral_region]
-        for index in range(0, len(spectral_region)):
-            # Instantiate merged regions
-            if not merged_regions:
-                merged_regions = [reg_as_tup[index]]
-            else:
-                last_merged = merged_regions[-1]
-                # If the lower value of the current subregion is less than or equal to the upper
-                # value of the last subregion added to merged_regions, update last_merged
-                # with the max upper value between the two regions.
-                if reg_as_tup[index][0] <= last_merged[1]:
-                    last_merged = (last_merged[0], max(last_merged[1], reg_as_tup[index][1]))
-                    merged_regions = merged_regions[:-1]
-                    merged_regions.append(last_merged)
-                else:
-                    merged_regions.append(reg_as_tup[index])
+        merged_regions = self.get_subsets(subset_name, spectral_only=True)
```

Comment:
Not sure how controversial this is, but in order to fix the XOR bug, I basically have to move the "merge overlapping regions" calls into `get_subsets` when `simplify_spectral=True`. Otherwise, if you apply XOR and then another operator like OR, you get overlaps even if you ask for `simplify_spectral=True`, because to fix the bug, I have to chop up regions in the XOR operations or it is going to throw out some regions that do not overlap with the XOR limits. Hope this is not too confusing.